File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0 || strs == null) {
            return "";
        }
        
        String longestCommonPrefix = strs[0];
        for (int index = 1; index < strs.length; index++) {
            int i = 0;
            while (i < longestCommonPrefix.length() && i < strs[index].length() &&
                  longestCommonPrefix.charAt(i) == strs[index].charAt(i)) {
                i++;
            }
            longestCommonPrefix = longestCommonPrefix.substring(0, i);
        }
        return longestCommonPrefix;
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        start = strs[0]
        end = strs[-1]
        i = 0
        j = 0
        prefix = ""

        while i < len(start) and j < len(end):
            if start[i] != end[j]:
                break
            else:
                prefix += start[i]
                i += 1
                j += 1

        
        return prefix
            